staging/src/k8s.io/kubectl/pkg/cmd/apply/apply_test.go

```diff
@@ -3219,3 +3298,247 @@ func TestApplySetDryRun(t *testing.T) {
+						_, ok := objMeta.Annotations[corev1.LastAppliedConfigAnnotation]
+						require.False(t, ok)
+						bodyRC := io.NopCloser(bytes.NewReader(noExistRC))
+						if p == serverSideAppliedPath {
+							bodyRC = io.NopCloser(bytes.NewReader(serverSideAppliedRC))
+						}
+						return &http.Response{StatusCode: http.StatusOK, Header: cmdtesting.DefaultHeader(), Body: bodyRC}, nil
+					default:
+						t.Errorf("unexpected request to %s", p)
+						return nil, nil
+					}
+				}),
+			}
+			tf.ClientConfigVal = cmdtesting.DefaultClientConfig()
+			ioStreams, _, outbuff, errBuf := genericclioptions.NewTestIOStreams()
+			cmdtesting.WithAlphaEnvs([]cmdutil.FeatureGate{cmdutil.AutoServerSide}, t, func(t *testing.T) {
+				cmd := NewCmdApply("kubectl", tf, ioStreams)
+				for _, filePath := range test.filePaths {
+					cmd.Flags().Set("filename", filePath)
+				}
+				cmd.Flags().Set("server-side", "auto")
+				cmd.Run(cmd, []string{})
+			})
+			require.Empty(t, errBuf.String())
+			assert.Equal(t, test.expectedOut, outbuff.String())
+			assert.Equal(t, test.expectedRequests.gets, getRequestCount, "unexpected number of GET requests")
+			assert.Equal(t, test.expectedRequests.applyPatches, applyPatchCount, "unexpected number of SSA patch requests")
+			assert.Equal(t, test.expectedRequests.smPatches, smpCount, "unexpected number of CSA patch requests")
+		})
+	}
+}
+
+func TestFieldMangerSetCorrectly(t *testing.T) {
+	cmdtesting.InitTestErrorHandler(t)
+	pathPrefix := "/namespaces/test/replicationcontrollers/"
+	serverSideAppliedName, serverSideAppliedRC := readReplicationController(t, filenameRCNoAnnotation)
+	serverSideAppliedPath := pathPrefix + serverSideAppliedName
+
+	clientSideAppliedName, clientSideAppliedRC := readReplicationController(t, filenameRCLASTAPPLIED)
+	clientSideAppliedPath := pathPrefix + clientSideAppliedName
+
+	isManagedName, isManagedRC := readReplicationController(t, filenameRC)
+	isManagedPath := pathPrefix + isManagedName
+
+	getRequestBodies := map[string]io.Reader{
+		serverSideAppliedPath: bytes.NewReader(serverSideAppliedRC),
+		clientSideAppliedPath: bytes.NewReader(clientSideAppliedRC),
+		isManagedPath:         bytes.NewReader(isManagedRC),
+	}
+
+	type requestExpectations struct {
+		applyPatches, smPatches int
+	}
+
+	tests := []struct {
+		name, filePath, fieldManager, expectedFieldManager string
+		expectedRequests                                   requestExpectations
+	}{
+		{
+			name:                 "default field manager on server-side apply",
+			filePath:             filenameRCNoAnnotation,
+			fieldManager:         "",
+			expectedFieldManager: fieldManagerServerSideApply,
+			expectedRequests: requestExpectations{
+				applyPatches: 1,
+				smPatches:    0,
+			},
+		},
+		{
+			name:                 "default field manager on client-side apply",
+			filePath:             filenameRC,
+			fieldManager:         "",
+			expectedFieldManager: FieldManagerClientSideApply,
```

Comment:
Oops, meant to remove it from all of them since with the collapsed patch we can't tell which are which - would you like me to add it back in?

Reply:
Since the main assertions are within the fake client, I would like to have some assertion proving that it was actually called. I'd be fine with either introducing a case statement on the patch type so that we can tell which is which, or having a static assertion (nothing in the test data--just set to zero after test.Run and checked at the end) that a patch request was sent.